File: cpp/164564.cpp
typedef long long ll;

ll powx(int x, int n, int mod) {
    if (n == 0)
        return 1ll;
    ll t = powx(x, n / 2, mod);
    t = t * t % mod;
    return n % 2 ? t * x % mod : t;
}

int sumOfBigSequence(int m, int n) {
    int mod = 1e9 + 7;
    ll x = m * (m - 1) / 2;
    ll y = m * m - 1;
    ll res = (x * ((powx(y + 1, n, mod) - 1 + mod) % mod) % mod + y) % mod;
    res = res * powx(y, mod - 2, mod) % mod;
    return res;
}

File: py/164564.py
# Lời giải từ bạn kien_the_sun

def sumOfBigSequence(m, n):
    #a[0] = 1
    #a[1] = (m*m-m) // 2
    #a[2] = a[1] * (m**2) 
    #a[3] = a[2] * (m**2)
    #.....
    mod = 10 ** 9 + 7
    return (1 +(m*m - m) * 500000004 * (pow(m**2, n, mod)-1)* pow(m**2-1, mod-2, mod)) % mod